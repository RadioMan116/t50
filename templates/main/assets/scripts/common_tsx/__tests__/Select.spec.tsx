import { mount, shallow } from 'enzyme';
import { h } from 'preact';
import { AssertionError } from 'assert';
import { shallowToJson } from 'enzyme-to-json';
import Select from '../src/Select';

global["$"] = global["jQuery"] = (function(){});

describe('select', () => {
    it('default initialize and click change', () => {
        let data = [{val: "1", title: "test1"}, {val: "2", title: "test2"}, {val: "3", title: "test3"}];
        let def = {val: "0", title: "-"}
        const wrapper = shallow(<Select items={data} default={def} onSelect={null} class="asda1das" value="2" />)
        let options = wrapper.find("option")
        expect(options.at(1).prop("selected")).toBe(false)
        expect(options.at(2).prop("selected")).toBe(true)
        expect(options.at(3).prop("selected")).toBe(false)

        // tslint:disable-block
        wrapper.setState({value: 3})
        options = wrapper.find("option")
        expect(options.at(0).prop("disabled")).toBe(true)
        expect(options.at(1).prop("selected")).toBe(false)
        expect(options.at(2).prop("selected")).toBe(false)
        expect(options.at(3).prop("selected")).toBe(true)
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('default initialize not disabled and click change', () => {
        let data = [{val: "1", title: "test1"}, {val: "2", title: "test2"}, {val: "3", title: "test3"}];
        let def = {val: 0, title: "-", disabled: false}
        const wrapper = shallow(<Select items={data} default={def} onSelect={null} class="asda1das" value="2" />)
        let options = wrapper.find("option")
        expect(options.at(0).prop("disabled")).toBe(false)
        expect(options.at(1).prop("selected")).toBe(false)
        expect(options.at(2).prop("selected")).toBe(true)
        expect(options.at(3).prop("selected")).toBe(false)

        // tslint:disable-block
        wrapper.setState({value: 0})
        options = wrapper.find("option")
        expect(options.at(0).prop("selected")).toBe(true)
        expect(options.at(1).prop("selected")).toBe(false)
        expect(options.at(2).prop("selected")).toBe(false)
        expect(options.at(3).prop("selected")).toBe(false)
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('select simple props set', () => {
        let data = [{val: "1", title: "test1"}, {val: "2", title: "test2"}, {val: "3", title: "test3"}];
        const wrapper = shallow(<Select items={data} onSelect={null} />)
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

});