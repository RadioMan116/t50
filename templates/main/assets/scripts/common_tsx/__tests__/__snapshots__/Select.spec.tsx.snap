// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select default initialize and click change 1`] = `
<select
  className="asda1das"
  disabled={false}
  value={3}
>
  <option
    disabled={true}
    selected={false}
    value="0"
  >
    -
  </option>
  <option
    disabled={false}
    selected={false}
    value="1"
  >
    test1
  </option>
  <option
    disabled={false}
    selected={false}
    value="2"
  >
    test2
  </option>
  <option
    disabled={false}
    selected={true}
    value="3"
  >
    test3
  </option>
</select>
`;

exports[`select default initialize not disabled and click change 1`] = `
<select
  className="asda1das"
  disabled={false}
  value={0}
>
  <option
    disabled={false}
    selected={true}
    value={0}
  >
    -
  </option>
  <option
    disabled={false}
    selected={false}
    value="1"
  >
    test1
  </option>
  <option
    disabled={false}
    selected={false}
    value="2"
  >
    test2
  </option>
  <option
    disabled={false}
    selected={false}
    value="3"
  >
    test3
  </option>
</select>
`;

exports[`select select simple props set 1`] = `
<select
  disabled={false}
>
  <option
    disabled={false}
    selected={false}
    value="1"
  >
    test1
  </option>
  <option
    disabled={false}
    selected={false}
    value="2"
  >
    test2
  </option>
  <option
    disabled={false}
    selected={false}
    value="3"
  >
    test3
  </option>
</select>
`;
